documentation/guide-usecase.asciidoc

```diff
@@ -53,23 +56,34 @@ public interface UcApproveBooking {
 ----
 
 == Implementation
-For the implementation of a use-cas, the same rules that are described for the link:guide-component-facade.asciidoc#implementation[component-facade implementation].
-
- .
-
-However, when following the use-case approach, your component facade simply changes to:
+For the implementation of a use-case, the same rules that are described for the link:guide-component-facade.asciidoc#implementation[component-facade implementation].
 
+In short, the implementation should carry its own name and the suffix `Impl` and is annotated with `@Named`. It will need access to the persistent data which is done by injecting the corresponding repository (or DAO). Furthermore, it shall not expose persistent entities from the data access layer and has to map them to transfer objects using the bean-mapper. Please refer to our bean mapping documentation for more information.
+Here is an example:
 [source,java]
 ----
-public interface Bookingmanagement extends UcFindBooking, UcManageBooking, UcApproveBooking {
+public class UcFindBookingImpl implements UcFindBooking {
+
+  @Inject
+  private BookingRepository bookingRepository;
+
+  @Override
+  public BookingEto findBooking(Long id) {
```

Comment:
Didn't we want to change from the concept with `Eto` and `Cto` objects to a simple concept with `Dto`s in the future? Then we should adjust the code examples in this guide and maybe add a link to an example with the old concept, so readers who only know the old style won't be confused

Reply:
That is a kind of tricky point.
To make our guide less biased, I would sugget to come up with a more general example that returns just any transfer-object (`*To`) and then for details refer to the `guide-transferobject.asciidoc`. This guide should again be copied to a `guide-eto-cto.asciidoc` and we can add a `guide-dto.asciidoc`. Then we can rewrite the `guide-transferobject.asciidoc` to only explain the general idea of transfer objects and then link to the two options `ETO/CTO` and `DTO` (please note that neither ETO is spring specific nor DTO is quarkus specific so do not put into these sub-folders).
As I am running out of time, could you file a story for this?